tls/s2n_connection.c

```diff
@@ -771,10 +771,12 @@ int s2n_connection_get_client_cert_chain(struct s2n_connection *conn, uint8_t **
     POSIX_ENSURE_REF(conn);
     POSIX_ENSURE_REF(der_cert_chain_out);
     POSIX_ENSURE_REF(cert_chain_len);
-    POSIX_ENSURE_REF(conn->secure.client_cert_chain.data);
+    POSIX_ENSURE_REF(conn->secure.peer_cert_chain.data);
 
-    *der_cert_chain_out = conn->secure.client_cert_chain.data;
-    *cert_chain_len = conn->secure.client_cert_chain.size;
+    POSIX_ENSURE(conn->mode == S2N_SERVER, S2N_ERR_INVALID_STATE);
+
+    *der_cert_chain_out = conn->secure.peer_cert_chain.data;
+    *cert_chain_len = conn->secure.peer_cert_chain.size;
```

Comment:
This is a refactor-- how are you making sure the behavior hasn't changed? Were there existing tests for this API?